salt/states/netyang.py

```diff
@@ -152,12 +153,16 @@ def managed(name,
     return_compliance_report = kwargs.get('compliance_report', False) or __opts__.get('compliance_report', False)
     profiles = kwargs.get('profiles', [])
     temp_file = __salt__['temp.file']()
-    log.debug('Creating temp file: {0}'.format(temp_file))
+    log.debug('Creating temp file: %s', temp_file)
     if 'to_dict' not in data:
         data = {'to_dict': data}
     data = [data]
     with salt.utils.files.fopen(temp_file, 'w') as file_handle:
-        salt.utils.yaml.safe_dump(salt.utils.json.loads(salt.utils.json.dumps(data)), file_handle, encoding='utf-8')
+        salt.utils.yaml.safe_dump(
+            salt.utils.json.loads(salt.utils.json.dumps(data)),
+            salt.utils.stringutils.to_str(file_handle),
```

Comment:
This will break, we're passing a filehandle to the dumper. This should just be `file_handle`, not `salt.utils.stringutils.to_str(file_handle)`.